### PR TITLE
Fix cursor position after completion

### PR DESCRIPTION
**Pull request by hrsh7th**
I noticed a bug of the completion.vim when writing rust.

It seems to place cursor for the wrong place at the timing of CompleteDone.

The VSCode will handle `CompleteItem.textEdit` and `CompleteItem.insertText` as a `snippet` so I aligned behavior to it.

### CODE REVIEW

File: autoload/lsp/ui/vim/completion.vim [hunk 1]
```diff
@@ -131,6 +131,7 @@ function! s:on_complete_done_after() abort
       endif
     else
       call lsp#utils#text_edit#apply_text_edits('%', [{ 'range': l:range, 'newText': l:text }])
+      call cursor(lsp#utils#position#lsp_to_vim('%', l:range.end))
     endif
```
Comment (hrsh7th) [inline autoload/lsp/ui/vim/completion.vim:135]: The snippet will place the cursor for to end of snippet (if it has no placeholders).
So we should place cursor to the range end if applying textEdit.

https://github.com/microsoft/vscode/blob/master/src/vs/editor/contrib/suggest/suggestController.ts#L374

**Comment by prabirshrestha** [comment]
LGTM.

can you add the comments in code. useful when revisiting the code in future during refactors or fixing other bugs.

**Comment by hrsh7th** [comment]
Thank you for your review.

I want to clarify the completion.vim algorithm but it hard...

**Comment by prabirshrestha** [comment]
This is actually causing me issues with cursors positioning.

```rust
fn main() {
    println!("Hello, world!");
}
```

Then below `println` type `PathBuf` once the autocomplete shows `std::path::PathBuf` press `:` . Then look at the cursor.

```rust
use std::path::PathBuf;

fn main() {
    println!("Hello, world!");
    PathBu|f
}
```

Currsor is before `f` instead of after `f`.

Wondering if it is off by 1 bug.

**Comment by hrsh7th** [comment]
It's reproduce on my side... I'm tacking to solve it.

**Pull request merged**